File: clients/javascript/src/models/queueMsgOut.ts
// this file is @generated

export interface QueueMsgOut {
    msgId: string;
    value: number[];
    headers?: { [key: string]: string };
    timestamp: Date;
}

export const QueueMsgOutSerializer = {
    _fromJsonObject(object: any): QueueMsgOut {
        return {
            msgId: object['msg_id'],
            value: object['value'],
            headers: object['headers'],
            timestamp: new Date(object['timestamp']),
        };
    },

    _toJsonObject(self: QueueMsgOut): any {
        return {
            'msg_id': self.msgId,
            'value': self.value,
            'headers': self.headers,
            'timestamp': self.timestamp,
        };
    }
}

File: z-clients/rust/src/models/queue_msg_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct QueueMsgOut {
    pub msg_id: String,

    #[serde(with = "serde_bytes")]
    pub value: Vec<u8>,

    pub headers: std::collections::HashMap<String, String>,

    #[serde(with = "crate::unix_timestamp_ms_serde")]
    pub timestamp: jiff::Timestamp,

    #[serde(
        with = "crate::unix_timestamp_ms_serde::optional",
        skip_serializing_if = "Option::is_none"
    )]
    pub scheduled_at: Option<jiff::Timestamp>,
}

impl QueueMsgOut {
    pub fn new(
        msg_id: String,
        value: Vec<u8>,
        headers: std::collections::HashMap<String, String>,
        timestamp: jiff::Timestamp,
    ) -> Self {
        Self {
            msg_id,
            value,
            headers,
            timestamp,
            scheduled_at: None,
        }
    }

    pub fn with_scheduled_at(mut self, value: impl Into<Option<jiff::Timestamp>>) -> Self {
        self.scheduled_at = value.into();
        self
    }
}
